gtsfm/utils/metrics.py

```diff
@@ -451,3 +453,70 @@ def compute_percentage_change(x: float, y: float) -> float:
         percentage change (may be positive or negative).
     """
     return (y - x) / (x + EPSILON) * 100
+
+
+def compute_psnr_after_downsampling(
+    original_point_cloud: np.ndarray, downsampled_point_cloud: np.ndarray
+) -> GtsfmMetric:
+    """Compute PSNR between original point cloud and downsampled point cloud
+    A larger PSNR shows that the downsampled point cloud consists better with the original point cloud
+
+    Args:
+        original_point_cloud: original dense point cloud before downsampling
+        downsampled_point_cloud: dense point cloud after downsampling
+
+    Returns:
+        GtsfmMetric: PSNR between original point cloud and downsampled point cloud
+    """
+    diagnose_voxel_scale = np.linalg.norm(mvs_utils.estimate_voxel_scales(original_point_cloud))
+    original_tree = KDTree(data=original_point_cloud)
```

Comment:
If you've seen this KDTree PSNR method used elsewhere, do you mind adding a reference @codyly ?